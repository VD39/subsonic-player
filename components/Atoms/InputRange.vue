<script setup lang="ts">
const props = defineProps<{
  buffer?: number;
  delay?: boolean;
  max: number;
  min: number;
}>();

const emit = defineEmits<{
  change: [value: number];
}>();

const internalModal = defineModel<number>();

const sliderRef = ref<HTMLElement | null>(null);
const thumbRef = ref<HTMLElement | null>(null);

const isSeeking = ref(false);
const isHovering = ref(false);
const hoverValue = ref(internalModal.value);
const pendingValue = ref(internalModal.value);

const progress = ref(getProgress(internalModal.value));
const bufferProgress = ref(getProgress(props.buffer));

const hoverProgress = computed(() => getProgress(hoverValue.value));
const isStandard = computed(() => !props.max);

function getProgress(newValue = 0) {
  if (!sliderRef.value) {
    return 0;
  }

  const { width: sliderWidth } = sliderRef.value.getBoundingClientRect();

  // If max is not set, set progress to be width of the slider.
  if (isStandard.value) {
    return sliderWidth;
  }

  return (newValue / props.max) * sliderWidth;
}

function handleModifyProgress(event: MouseEvent | TouchEvent) {
  if (!sliderRef.value) {
    return;
  }

  const { left, width } = sliderRef.value.getBoundingClientRect();

  const pageX = 'pageX' in event ? event.pageX : event.changedTouches[0].pageX;

  const processedValue = ((pageX - left) / width) * props.max;
  const clippedValue = Math.min(Math.max(processedValue, props.min), props.max);

  hoverValue.value = clippedValue;

  if (isSeeking.value) {
    pendingValue.value = clippedValue;
    progress.value = getProgress(pendingValue.value);

    // Don't update value when scrolling.
    if (!props.delay) {
      updateValue();
    }
  }
}

function updateValue() {
  internalModal.value = pendingValue.value;
  emit('change', pendingValue.value!);
}

function onMouseMove(event: MouseEvent | TouchEvent) {
  isHovering.value = false;
  handleModifyProgress(event);
}

function onMouseUp() {
  updateValue();
  document.removeEventListener('mouseup', onMouseUp);
  document.removeEventListener('mousemove', onMouseMove);

  document.removeEventListener('touchend', onMouseUp);
  document.removeEventListener('touchmove', onMouseMove);
  isSeeking.value = false;
}

function onSliderMouseDown(event: MouseEvent | TouchEvent) {
  isSeeking.value = true;
  handleModifyProgress(event);

  document.addEventListener('mouseup', onMouseUp);
  document.addEventListener('mousemove', onMouseMove);

  document.addEventListener('touchend', onMouseUp, {
    passive: true,
  });
  document.addEventListener('touchmove', onMouseMove, {
    passive: true,
  });
}

function onSliderMouseOver() {
  isHovering.value = true;
}

function onSliderMouseMove(event: MouseEvent) {
  if (isHovering.value) {
    handleModifyProgress(event);
  }
}

function updateProgress() {
  // Only update when user is not seeking.
  if (!isSeeking.value) {
    progress.value = getProgress(internalModal.value);
  }

  if (props.buffer) {
    bufferProgress.value = getProgress(props.buffer);
  }
}

watch(() => [props.buffer, internalModal.value], updateProgress, {
  immediate: true,
});

const onResize = debounce(updateProgress);

onMounted(() => {
  updateProgress();

  window.addEventListener('resize', onResize);
});

onUnmounted(() => {
  window.removeEventListener('resize', onResize);
});
</script>

<template>
  <div
    :class="[
      $style.inputRange,
      {
        [$style.seeking]: isSeeking,
        [$style.standard]: isStandard,
      },
    ]"
  >
    <div
      ref="sliderRef"
      :class="['centerItems', $style.slider]"
      @mousedown.stop.prevent="onSliderMouseDown"
      @mousemove="onSliderMouseMove"
      @mouseover="onSliderMouseOver"
      @touchstart.stop.passive="onSliderMouseDown"
    >
      <div :class="$style.progressWrapper">
        <div
          ref="progressBar"
          :class="$style.process"
          :style="{ width: `${progress}px` }"
        />

        <div
          v-if="buffer"
          ref="bufferBar"
          :class="$style.buffer"
          :style="{ width: `${bufferProgress}px` }"
        />
      </div>

      <div
        v-if="!isStandard"
        ref="thumbRef"
        :class="$style.thumb"
        :style="{ left: `${progress - 6}px` }"
        @mousedown.stop.prevent="onSliderMouseDown"
        @touchstart.stop.passive="onSliderMouseDown"
      />

      <div
        v-if="$slots.default && !isStandard"
        ref="tooltip"
        :class="['mBS', 'strong', 'smallFont', $style.tooltip]"
        :style="{ left: `${hoverProgress}px` }"
      >
        <slot :pending-value="hoverValue" />
      </div>
    </div>
  </div>
</template>

<style module>
.inputRange {
  position: relative;
  width: var(--width-height-100);
  padding: var(--default-space) 0;

  @media (hover: hover) {
    &.seeking,
    &:hover {
      .tooltip {
        --tooltip-opacity: 1;
      }
    }
  }
}

.slider {
  width: var(--width-height-100);
  height: 6px;
  cursor: pointer;
  background-color: var(--invert-color);

  .standard & {
    pointer-events: none;
    cursor: unset;
  }
}

.progressWrapper {
  --progress-wrapper-width-height: var(--width-height-100);

  position: relative;
  width: var(--progress-wrapper-width-height);
  height: var(--progress-wrapper-width-height);
  overflow: hidden;
}

.buffer,
.process {
  position: absolute;
  inset: 0 auto auto 0;
  z-index: 1;
  height: var(--width-height-100);
  background-color: var(--theme-color);

  .standard & {
    --process-background-color: color-mix(
      in oklab,
      var(--body-font-color) 25%,
      transparent
    );

    background-image: linear-gradient(
      -45deg,
      var(--process-background-color) 25%,
      transparent 25%,
      transparent 50%,
      var(--process-background-color) 50%,
      var(--process-background-color) 75%,
      transparent 75%,
      transparent
    );
    background-size: 50px 50px;
    animation: stripes-move 10s linear infinite;
  }
}

.buffer {
  z-index: 0;
  background-color: var(--theme-color);
  opacity: 0.5;
}

.thumb {
  --thumb-width-height: 12px;

  position: absolute;
  z-index: 2;
  display: block;
  width: var(--thumb-width-height);
  height: var(--thumb-width-height);
  cursor: pointer;
  background-color: var(--theme-color);
  border-radius: var(--border-radius-round);
  box-shadow: var(--box-shadow-medium);
}

.tooltip {
  --tooltip-opacity: 0;
  --tooltip-background-color: var(--body-background-color);

  position: absolute;
  inset: auto auto var(--space-32) 50%;
  z-index: 9;
  padding: var(--space-4);
  color: var(--body-font-color);
  white-space: nowrap;
  pointer-events: none;
  background-color: var(--tooltip-background-color);
  box-shadow: var(--box-shadow-large);
  opacity: var(--tooltip-opacity);
  transition:
    transform var(--transition),
    opacity var(--transition);
  transform: translate(-50%, 10px) scale(0.8);
  transform-origin: 50% 100%;

  &::before {
    position: absolute;
    inset: 100% auto auto 50%;
    margin-left: -5px;
    content: '';
    border-color: var(--tooltip-background-color) transparent transparent
      transparent;
    border-style: solid;
    border-width: 5px;
  }

  @media (--tablet-up) {
    bottom: var(--space-24);
  }
}

@keyframes stripes-move {
  0% {
    background-position: 0 0;
  }

  100% {
    background-position: 50px 50px;
  }
}
</style>
