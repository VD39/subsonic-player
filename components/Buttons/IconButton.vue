<script setup lang="ts">
const props = withDefaults(
  defineProps<{
    disabled?: boolean;
    fullWidth?: boolean;
    icon?: string;
    iconColor?: string;
    iconPosition?: IconPosition;
    iconSize?: number;
    iconWeight?: IconWeight;
    is?: ButtonType;
    showText?: boolean;
    type?: string;
  }>(),
  {
    disabled: undefined,
    icon: undefined,
    iconColor: undefined,
    iconPosition: 'left',
    iconSize: 24,
    iconWeight: 'duotone',
    is: 'button',
    type: 'button',
  },
);

const type = props.is === 'button' ? props.type : undefined;
const isComponent =
  props.is === 'nuxt-link' ? resolveComponent('NuxtLink') : props.is;
</script>

<template>
  <component
    :is="isComponent"
    :class="[
      $style.iconButton,
      {
        [$style.alignRight]: icon && iconPosition === 'right',
        [$style.disabled]: disabled,
        [$style.fullWidth]: fullWidth,
      },
    ]"
    :disabled="disabled"
    :type="type"
    v-bind="$attrs"
  >
    <component
      :is="icon"
      v-if="icon"
      ref="phIcon"
      :class="$style.icon"
      aria-hidden="true"
      :size="iconSize"
      :weight="iconWeight"
      :color="iconColor"
    />

    <span
      ref="text"
      :class="[
        $style.text,
        {
          'visually-hidden': !showText,
        },
      ]"
    >
      <slot />
    </span>
  </component>
</template>

<style module>
.iconButton {
  @mixin align-center;

  position: relative;
  display: inline-flex;
  gap: var(--space-8);
  padding: var(--button-spacing);
  margin: 0;
  white-space: nowrap;
}

.alignRight {
  flex-direction: row-reverse;
}

.fullWidth {
  width: 100%;
}

.disabled {
  pointer-events: none;
  opacity: 0.25;
}

.icon {
  flex-shrink: 0;

  [opacity='0.2'] {
    opacity: 0.4;
    fill: var(--secondary-font-color);
  }
}

.text {
  @mixin ellipsis-overflow;

  display: block;
  margin-bottom: 1px;
}
</style>
