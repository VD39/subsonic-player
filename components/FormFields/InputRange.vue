<script setup lang="ts">
const props = defineProps<{
  buffer?: number;
  delay?: boolean;
  hideThumb?: boolean;
  max: number;
  min: number;
  modelValue: number;
}>();

const emit = defineEmits(['update:modelValue', 'change']);

const value = computed({
  get() {
    return props.modelValue;
  },
  set(newValue) {
    emit('update:modelValue', newValue);
  },
});

const sliderRef = ref<HTMLElement | null>(null);
const thumbRef = ref<HTMLElement | null>(null);

const isSeeking = ref(false);
const isHovering = ref(false);
const hoverValue = ref(props.modelValue);
const pendingValue = ref(props.modelValue);

const progress = ref(getProgress(props.modelValue));
const bufferProgress = ref(getProgress(props.buffer));

const hoverProgress = computed(() => getProgress(hoverValue.value));

function getProgress(newValue = 0) {
  if (!sliderRef.value) {
    return 0;
  }

  return (newValue / props.max) * sliderRef.value.getBoundingClientRect().width;
}

function handleModifyProgress(event: MouseEvent) {
  if (!sliderRef.value) {
    return;
  }

  const { left, width } = sliderRef.value.getBoundingClientRect();

  const processedValue = ((event.pageX - left) / width) * props.max;
  const clippedValue = Math.min(Math.max(processedValue, props.min), props.max);

  hoverValue.value = clippedValue;

  if (isSeeking.value) {
    pendingValue.value = clippedValue;
    progress.value = getProgress(pendingValue.value);

    // don't update value when scrolling.
    if (!props.delay) {
      updateValue();
    }
  }
}

function updateValue() {
  value.value = pendingValue.value;
  emit('change', pendingValue.value);
}

function onMouseMove(event: MouseEvent) {
  isHovering.value = false;
  handleModifyProgress(event);
}

function onMouseUp() {
  updateValue();
  document.removeEventListener('mouseup', onMouseUp);
  document.removeEventListener('mousemove', onMouseMove);
  isSeeking.value = false;
}

function onSliderMouseDown(event: MouseEvent) {
  isSeeking.value = true;
  handleModifyProgress(event);

  document.addEventListener('mouseup', onMouseUp);
  document.addEventListener('mousemove', onMouseMove);
}

function onSliderMouseOver() {
  isHovering.value = true;
}

function onSliderMouseMove(event: MouseEvent) {
  if (isHovering.value) {
    handleModifyProgress(event);
  }
}

function updateProgress() {
  // Only update when user is not seeking.
  if (!isSeeking.value) {
    progress.value = getProgress(props.modelValue);
  }

  if (props.buffer) {
    bufferProgress.value = getProgress(props.buffer);
  }
}

watch(() => [props.buffer, props.modelValue], updateProgress, {
  immediate: true,
});

function onResize() {
  updateProgress();
}

onMounted(() => {
  updateProgress();

  window.addEventListener('resize', onResize);
});

onUnmounted(() => {
  window.removeEventListener('resize', onResize);
});
</script>

<template>
  <div
    :class="[
      $style.sliderWrapper,
      {
        [$style.seeking]: isSeeking,
      },
    ]"
    v-bind="$attrs"
  >
    <div
      ref="sliderRef"
      :class="$style.slider"
      @mousedown.stop.prevent="onSliderMouseDown"
      @mouseover="onSliderMouseOver"
      @mousemove="onSliderMouseMove"
    >
      <div :class="$style.progressWrapper">
        <div
          ref="progressBar"
          :class="$style.process"
          :style="{ width: `${progress}px` }"
        />

        <div
          v-if="buffer"
          ref="bufferBar"
          :class="$style.buffer"
          :style="{ width: `${bufferProgress}px` }"
        />
      </div>

      <div
        v-if="!hideThumb"
        ref="thumbRef"
        :class="$style.thumb"
        :style="{ left: `${progress - 6}px` }"
        @mousedown.stop.prevent="onSliderMouseDown"
      />

      <div
        v-if="$slots.default"
        ref="defaultSlot"
        :class="$style.tooltip"
        :style="{ left: `${hoverProgress}px` }"
      >
        <slot :pending-value="hoverValue" />
      </div>
    </div>
  </div>
</template>

<style module>
.tooltip {
  position: absolute;
  inset: auto auto var(--space-12) 50%;
  z-index: 9;
  padding: var(--space-4);
  margin-bottom: var(--space-8);
  font-size: var(--small-font-size);
  font-weight: 600;
  color: var(--black-color);
  white-space: nowrap;
  pointer-events: none;
  background-color: var(--white-color);
  box-shadow: var(--box-shadow-large);
  opacity: 0;
  transition:
    transform 0.2s ease 0.1s,
    opacity 0.2s ease 0.1s;
  transform: translate(-50%, 10px) scale(0.8);
  transform-origin: 50% 100%;

  &::before {
    position: absolute;
    inset: auto auto calc(var(--space-4) * -1) 50%;
    z-index: 9;
    width: 0;
    height: 0;
    content: '';
    border-top: 4px solid var(--white-color);
    border-right: 4px solid var(--black-color);
    border-left: 4px solid var(--black-color);
    transform: translateX(-50%);
  }
}

.sliderWrapper {
  padding: var(--space-8) 0;

  &.seeking,
  &:hover,
  &:focus {
    .tooltip {
      opacity: 1;
    }
  }
}

.slider {
  @mixin align-center;

  position: relative;
  width: 100%;
  height: 6px;
  cursor: pointer;
  background-color: var(--invert-color);
}

.progressWrapper {
  position: relative;
  width: 100%;
  height: 100%;
  overflow: hidden;
}

.buffer,
.process {
  position: absolute;
  inset: 0 auto auto 0;
  z-index: 1;
  height: 100%;
  background-color: var(--theme-color);
}

.buffer {
  z-index: 0;
  background-color: var(--theme-color);
  opacity: 0.5;
}

.thumb {
  position: absolute;
  z-index: 2;
  display: block;
  width: 12px;
  height: 12px;
  cursor: pointer;
  background-color: var(--theme-color);
  border-radius: 50%;
  box-shadow: var(--box-shadow-medium);
}
</style>
