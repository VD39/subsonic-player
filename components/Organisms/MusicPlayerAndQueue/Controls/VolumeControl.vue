<script setup lang="ts">
import ButtonLink from '@/components/Atoms/ButtonLink.vue';
import InputRange from '@/components/Atoms/InputRange.vue';

const { isMuted, setVolume, toggleMute, volume } = useAudioPlayer();

const ariaValueText = computed(() => `${volume.value} of 1`);
const volumePercentage = computed(() => Math.floor(volume.value * 100));
const volumeTitle = computed(() => (isMuted.value ? 'Unmute' : 'Mute'));
const volumeIcon = computed(() => {
  switch (true) {
    case volume.value > 0.5:
      return ICONS.volume05;
    case volume.value > 0.2:
      return ICONS.volume02;
    case volume.value > 0:
      return ICONS.volume0;
    default:
      return ICONS.volumeDefault;
  }
});

function changeAudioVolume() {
  setVolume(volume.value);
}
</script>

<template>
  <div :class="['centerItems', $style.volumeControl]">
    <ButtonLink :icon="volumeIcon" :title="volumeTitle" @click="toggleMute">
      {{ volumeTitle }}
    </ButtonLink>

    <div :class="['centerItems', 'smallFont', $style.volumeInputWrapper]">
      <InputRange
        v-model="volume"
        aria-label="Volume"
        :aria-valuetext="ariaValueText"
        :max="1"
        :min="0"
        @change="changeAudioVolume"
      />

      <p aria-hidden :class="$style.currentVolume">{{ volumePercentage }}</p>

      <p ref="currentVolume" class="visuallyHidden">
        Volume at {{ volumePercentage }}%
      </p>
    </div>
  </div>
</template>

<style module>
.volumeControl {
  position: relative;

  @media (hover: hover) {
    &:hover {
      .volumeInputWrapper {
        --volume-input-visibility: visible;
      }
    }
  }
}

.volumeInputWrapper {
  --volume-input-visibility: hidden;

  position: absolute;
  top: calc(var(--default-space) * -4);
  left: 50%;
  z-index: 10;
  visibility: var(--volume-input-visibility);
  gap: var(--default-space);
  width: 200px;
  padding: var(--default-space);
  background: var(--secondary-background-color);
  border: 1px solid var(--border-color);
  border-radius: var(--border-radius-medium);
  box-shadow: var(--box-shadow-large);
  transform: translateX(-50%);
  transition: visibility var(--transition);
}

.currentVolume {
  width: 25px;
  text-align: right;
}
</style>
