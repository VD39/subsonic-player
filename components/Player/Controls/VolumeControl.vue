<script setup lang="ts">
import IconButton from '@/components/Buttons/IconButton.vue';
import InputRange from '@/components/FormFields/InputRange.vue';

const { setVolume, isMuted, toggleVolume, volume } = useAudioPlayer();

const previousVolume = ref(volume.value);

const ariaValueText = computed(() => `${volume.value} of 1`);
const volumeTitle = computed(() => (isMuted.value ? 'Unmute' : 'Mute'));
const volumeIcon = computed(() => {
  if (volume.value > 0.5) {
    return 'PhSpeakerHigh';
  }

  if (volume.value > 0.2) {
    return 'PhSpeakerLow';
  }

  if (volume.value > 0) {
    return 'PhSpeakerNone';
  }

  return 'PhSpeakerX';
});

function changeAudioVolume() {
  previousVolume.value = volume.value;
  setVolume(volume.value);
}

function toggleAudioVolume() {
  toggleVolume(previousVolume.value);
}
</script>

<template>
  <div :class="$style.volumeControl">
    <IconButton
      :icon="volumeIcon"
      :title="volumeTitle"
      @click="toggleAudioVolume"
    >
      {{ volumeTitle }}
    </IconButton>

    <div :class="$style.volumeInputWrapper">
      <InputRange
        v-model="volume"
        :min="0"
        :max="1"
        aria-label="Volume"
        :class="$style.volumeInput"
        :aria-valuetext="ariaValueText"
        hide-thumb
        @change="changeAudioVolume"
      />

      <p ref="currentVolume" :class="$style.currentVolume">
        <span class="visually-hidden">Volume at </span>
        {{ Math.floor(volume * 100) }}
        <span class="visually-hidden">%</span>
      </p>
    </div>
  </div>
</template>

<style module>
.volumeControl {
  @mixin align-center;

  position: relative;

  &:hover {
    .volumeInputWrapper {
      visibility: visible;
    }
  }
}

/* stylelint-disable-next-line no-descending-specificity */
.volumeInputWrapper {
  @mixin align-center;

  position: absolute;
  top: calc(var(--space-12) * -3);
  left: 50%;
  z-index: 5;
  gap: var(--space-8);
  width: 200px;
  padding: var(--space-8) var(--space-12);
  font-size: var(--small-font-size);
  visibility: hidden;
  background: var(--secondary-background-color);
  border: 1px solid var(--border-color);
  border-radius: var(--border-radius-medium);
  box-shadow: var(--box-shadow-large);
  transition: visibility 0.5s ease-in-out;
  transform: translateX(-50%);
}

.volumeInput {
  width: 100%;
}

.currentVolume {
  width: 20px;
  text-align: center;
}
</style>
