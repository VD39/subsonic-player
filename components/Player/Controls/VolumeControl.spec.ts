import type { VueWrapper } from '@vue/test-utils';
import { mount } from '@vue/test-utils';
import { useAudioPlayerMock } from '@/test/useAudioPlayerMock';
import IconButton from '@/components/Buttons/IconButton.vue';
import InputRange from '@/components/FormFields/InputRange.vue';
import VolumeControl from './VolumeControl.vue';

const { setVolumeMock, isMutedMock, toggleVolumeMock, volumeMock } =
  useAudioPlayerMock();

function factory(props = {}) {
  return mount(VolumeControl, {
    props: {
      ...props,
    },
  });
}

describe('VolumeControl', () => {
  let wrapper: VueWrapper;

  beforeEach(() => {
    wrapper = factory();
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('when volume button is clicked', () => {
    beforeEach(() => {
      wrapper.findComponent(IconButton).vm.$emit('click');
    });

    it('calls the toggleVolume function', () => {
      expect(toggleVolumeMock).toHaveBeenCalled();
    });
  });

  describe('when input range is changed', () => {
    beforeEach(() => {
      wrapper.findComponent(InputRange).vm.$emit('change');
    });

    it('calls the setVolume function', () => {
      expect(setVolumeMock).toHaveBeenCalled();
    });
  });

  describe('when isMuted is false', () => {
    it('sets the correct title on button component', () => {
      expect(wrapper.findComponent(IconButton).attributes('title')).toBe(
        'Mute',
      );
    });
  });

  describe('when isMuted is true', () => {
    beforeEach(() => {
      isMutedMock.value = true;
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('sets the correct title on button component', () => {
      expect(wrapper.findComponent(IconButton).attributes('title')).toBe(
        'Unmute',
      );
    });
  });

  describe.each([
    [0, 'PhSpeakerX'],
    [0.1, 'PhSpeakerNone'],
    [0.3, 'PhSpeakerLow'],
    [0.6, 'PhSpeakerHigh'],
  ])('when volume is %f', (volume, icon) => {
    beforeEach(() => {
      volumeMock.value = volume;
    });

    it('matches the snapshot', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });

    it('sets the correct icon on button component', () => {
      expect(wrapper.findComponent(IconButton).props('icon')).toBe(icon);
    });

    it('sets the correct aria-valuetext on the input range', () => {
      expect(
        wrapper.findComponent(InputRange).attributes('aria-valuetext'),
      ).toBe(`${volume} of 1`);
    });

    it('sets the correct current volume percentage', () => {
      expect(wrapper.find({ ref: 'currentVolume' }).text()).toBe(
        `Volume at ${volume * 100}%`,
      );
    });
  });
});
