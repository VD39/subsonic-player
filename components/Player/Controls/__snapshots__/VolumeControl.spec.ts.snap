// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControl > matches the snapshot 1`] = `
"<div class="volumeControl"><button class="iconButton" type="button" title="Mute">
    <phspeakerhigh class="icon" aria-hidden="true" size="24" weight="duotone"></phspeakerhigh><span class="text visually-hidden">Mute</span>
  </button>
  <div class="volumeInputWrapper">
    <div class="sliderWrapper volumeInput volumeInput" aria-label="Volume" aria-valuetext="1 of 1">
      <div class="slider">
        <div class="progressWrapper">
          <div class="process" style="width: 0px;"></div>
          <!--v-if-->
        </div>
        <!--v-if-->
        <!--v-if-->
      </div>
    </div>
    <p class="visually-hidden"> Volume at 100% </p>
  </div>
</div>"
`;

exports[`VolumeControl > when isMuted is true > matches the snapshot 1`] = `
"<div class="volumeControl"><button class="iconButton" type="button" title="Unmute">
    <phspeakerhigh class="icon" aria-hidden="true" size="24" weight="duotone"></phspeakerhigh><span class="text visually-hidden">Unmute</span>
  </button>
  <div class="volumeInputWrapper">
    <div class="sliderWrapper volumeInput volumeInput" aria-label="Volume" aria-valuetext="1 of 1">
      <div class="slider">
        <div class="progressWrapper">
          <div class="process" style="width: 0px;"></div>
          <!--v-if-->
        </div>
        <!--v-if-->
        <!--v-if-->
      </div>
    </div>
    <p class="visually-hidden"> Volume at 100% </p>
  </div>
</div>"
`;

exports[`VolumeControl > when volume is 0 > matches the snapshot 1`] = `
"<div class="volumeControl"><button class="iconButton" type="button" title="Unmute">
    <phspeakerx class="icon" aria-hidden="true" size="24" weight="duotone"></phspeakerx><span class="text visually-hidden">Unmute</span>
  </button>
  <div class="volumeInputWrapper">
    <div class="sliderWrapper volumeInput volumeInput" aria-label="Volume" aria-valuetext="0 of 1">
      <div class="slider">
        <div class="progressWrapper">
          <div class="process" style="width: 0px;"></div>
          <!--v-if-->
        </div>
        <!--v-if-->
        <!--v-if-->
      </div>
    </div>
    <p class="visually-hidden"> Volume at 0% </p>
  </div>
</div>"
`;

exports[`VolumeControl > when volume is 0.1 > matches the snapshot 1`] = `
"<div class="volumeControl"><button class="iconButton" type="button" title="Unmute">
    <phspeakernone class="icon" aria-hidden="true" size="24" weight="duotone"></phspeakernone><span class="text visually-hidden">Unmute</span>
  </button>
  <div class="volumeInputWrapper">
    <div class="sliderWrapper volumeInput volumeInput" aria-label="Volume" aria-valuetext="0.1 of 1">
      <div class="slider">
        <div class="progressWrapper">
          <div class="process" style="width: 0px;"></div>
          <!--v-if-->
        </div>
        <!--v-if-->
        <!--v-if-->
      </div>
    </div>
    <p class="visually-hidden"> Volume at 10% </p>
  </div>
</div>"
`;

exports[`VolumeControl > when volume is 0.3 > matches the snapshot 1`] = `
"<div class="volumeControl"><button class="iconButton" type="button" title="Unmute">
    <phspeakerlow class="icon" aria-hidden="true" size="24" weight="duotone"></phspeakerlow><span class="text visually-hidden">Unmute</span>
  </button>
  <div class="volumeInputWrapper">
    <div class="sliderWrapper volumeInput volumeInput" aria-label="Volume" aria-valuetext="0.3 of 1">
      <div class="slider">
        <div class="progressWrapper">
          <div class="process" style="width: 0px;"></div>
          <!--v-if-->
        </div>
        <!--v-if-->
        <!--v-if-->
      </div>
    </div>
    <p class="visually-hidden"> Volume at 30% </p>
  </div>
</div>"
`;

exports[`VolumeControl > when volume is 0.6 > matches the snapshot 1`] = `
"<div class="volumeControl"><button class="iconButton" type="button" title="Unmute">
    <phspeakerhigh class="icon" aria-hidden="true" size="24" weight="duotone"></phspeakerhigh><span class="text visually-hidden">Unmute</span>
  </button>
  <div class="volumeInputWrapper">
    <div class="sliderWrapper volumeInput volumeInput" aria-label="Volume" aria-valuetext="0.6 of 1">
      <div class="slider">
        <div class="progressWrapper">
          <div class="process" style="width: 0px;"></div>
          <!--v-if-->
        </div>
        <!--v-if-->
        <!--v-if-->
      </div>
    </div>
    <p class="visually-hidden"> Volume at 60% </p>
  </div>
</div>"
`;
